OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -73,6 +71,12 @@ protected virtual Target RecalculateTarget(Actor self, out bool targetIsHiddenAc
 
 		public override Activity Tick(Actor self)
 		{
+			if (ChildActivity != null)
+			{
+				ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
```

Comment:
In many cases, like this one, we should pre-tick the child activity to avoid a single-tick pause in the parent.
In my recent PRs I have been handling this by queuing `ActivityUtils.RunActivity(<inner>)`, but this probably wants to move into a `pretick = true` argument on `Activity.QueueChild` (which this should also be using).

Reply:
Implemented optional pre-tick in `QueueChild`.